packages/@aws-cdk/aws-codedeploy/test/server/test.deployment-group.ts

```diff
@@ -28,6 +28,21 @@ export = {
       test.done();
     },
 
+    'creating an application with physical name if needed'(test: Test) {
+      const stack = new cdk.Stack();
+      new codedeploy.ServerDeploymentGroup(stack, 'MyDG', {
+        deploymentGroupName: cdk.PhysicalName.GENERATE_IF_NEEDED,
+      });
+
+      expect(stack).to(haveResource('AWS::CodeDeploy::DeploymentGroup', {
```

Comment:
You need to be asserting something about the `AWS::CodeDeploy::Application`, not the DeploymentGroup.

Reply:
Yuh, but there won't be a field called `ApplicationName` in the output. That fails the build.

Reply:
Then the issue is not fixed.

Reply:
```
- Field ApplicationName missing in:
    {
      "Type": "AWS::CodeDeploy::Application",
      "Properties": {
        "ComputePlatform": "Server"
      }
    }
```
output for the test case with the resource type `AWS::CodeDeploy:;DeploymentGroup`. Same output will be received when the resource type of `can be created by explicitly passing an Application` test changed